lib/AST/ASTPrinter.cpp

```diff
@@ -3602,6 +3602,10 @@ static bool usesFeatureExtern(Decl *decl) {
   return decl->getAttrs().hasAttribute<ExternAttr>();
 }
 
+static bool usesFeatureStaticExclusiveOnly(Decl *decl) {
+  return decl->getAttrs().hasAttribute<StaticExclusiveOnlyAttr>();
```

Comment:
This would mean that declarations with new attributes are guarded by `#if` but what about uses of such declarations i.e. if this `decl` is i.e. a type that has a member that refers to a declaration with the attribute it would have to be wrapped in `#if` too, right?

Reply:
No, because in this example:
```swift
@_newAttr
struct A {}

struct B {
  let a: A
}
```
old compilers can understand `B` but if they don't know about the attribute that needs to be conditionalized. (There are lots of other similar types of attributes doing the same in this file)